src/test/java/com/scalyr/integrations/kafka/AddEventsClientTest.java

```diff
@@ -420,14 +422,24 @@ public void testTooLargeAddEventsSkipped() throws Exception {
     AddEventsClient addEventsClient = new AddEventsClient(scalyrUrl, API_KEY_VALUE, ADD_EVENTS_TIMEOUT_MS, ADD_EVENTS_RETRY_DELAY_MS, compressor);
     List<Event> events = createTestEvents(numEvents, numServers, numLogFiles, numParsers);
     events.forEach(event -> event.setMessage(largeMsg));
-    addEventsClient.log(events);
+
+    assertEquals(0, server.getRequestCount());
+
+    AddEventsResponse addEventsResponse = addEventsClient.log(events).get(5, TimeUnit.SECONDS);;
+    assertEquals("success", addEventsResponse.getStatus());
+    assertEquals("Skipped due to payload too large", addEventsResponse.getMessage());
 
     // request should be skipped
     assertEquals(0, server.getRequestCount());
 
     // Send next batch that is smaller than max payload size
     events = createTestEvents(numEvents, numServers, numLogFiles, numParsers);
-    addEventsClient.log(events);
+    addEventsResponse = addEventsClient.log(events).get(5, TimeUnit.SECONDS);
```

Comment:
Should pass in completable future from previous add events request by calling `AddEventsClient.log(List<Event> events, @Nullable CompletableFuture<AddEventsResponse> dependentAddEvents)`.  As part of the pipelining, we wait for the previous add events call to complete before issuing the next add events call.  This more closely mimics that the `ScalyrSinkTask` does.  I may not have done that earlier myself.

Reply:
I checked the code and we only do that if ``dependentAddEvents`` is not null so those tests are probably fine for now?

Reply:
Yes, it will work with `null`.  However, what we want to test is that the result of an Add events call that is skipped does not cause any problems with subsequent calls.